base/mqueue/platform_event.go

```diff
@@ -131,21 +138,14 @@ func (data *InventoryAIDs) WriteEvents(ctx context.Context, w Writer) error {
 	return err
 }
 
-func (data *EvalDataSlice) WriteEvents(ctx context.Context, w Writer) error {
-	batches, groupedSys, groupedReqID, orgIDMap := data.groupData()
-	groupedSysVal := groupedData{}
-	groupedReqIDVal := groupedData{}
-	if groupedSys != nil {
-		groupedSysVal = *groupedSys
-	}
-	if groupedReqID != nil {
-		groupedReqIDVal = *groupedReqID
-	}
+func (evals *EvalDataSlice) WriteEvents(ctx context.Context, w Writer) error {
+	// invs, reqs, orgIDs never nil
```

Comment:
what if someone will return nil in `getAccountEvalData` in the future? It's still a valid return value.
Can't we make `reqs=nil` and use EvalDataSlice instead of the InventoryAccounts? `reqs`'s the only difference in those 2 `WriteEvents` functions. Just a thought

Reply:
I refactored this because it was mixed approach of assuming dereferencing is one time safe and other is not, while they all are the same, never nil pointers, so lack of uniformity.

Now it's better because if you handle nil pointer inside getter function, you don't repeat itself every time you use it and do not obfuscate other functions. It's very popular to handle nil pointers in getters and return default value if it's nil - empty struct in this case.

If one changes getter, he ought to check it's receivers.

Mayve EvalDataSlice can be reused indeed. I'll take a look.

Reply:
Slice in Go is just
```
type SliceHeader struct {
	Data [uintptr](https://pkg.go.dev/builtin#uintptr)
	Len  [int](https://pkg.go.dev/builtin#int)
	Cap  [int](https://pkg.go.dev/builtin#int)
}
```
https://pkg.go.dev/reflect#SliceHeader
So all the heavy stuff is pointer anyway here, so probably it will be faster to return from these getters as value, and we will get rid of nil pointer issue :slightly_smiling_face:

Reply:
It's the same performance
```
By value
    1672            646303 ns/op
By reference
    1602            658101 ns/op
```
Let's just drop pointers from getters.

Reply:
I agree with passing by values, that's a good point. Please try to look if it is possible to use only `EvalDataSlice`